libcudacxx/include/cuda/std/__mdspan/mdspan.h

```diff
@@ -108,28 +113,44 @@ class mdspan
       return (__self.rank() > 0)
           && __MDSPAN_FOLD_OR((__self.__mapping_ref().extents().template __extent<_Idxs>() == index_type(0)));
     }
+
+    template <class... _SizeTypes>
+    _CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI static constexpr bool
+    __check_index(_Extents const& exts, _SizeTypes... __indices)
+    {
+      // std::array supports zero size
+      // ternary avoids "pointless comparison of unsigned integer with zero" warning
+      array<bool, sizeof...(_SizeTypes)> __res{
+        ((is_unsigned_v<index_type> ? true : static_cast<index_type>(__indices) >= 0)
+         && static_cast<index_type>(__indices) < exts.extent(_Idxs))...};
+      return _CUDA_VSTD::all_of(__res.begin(), __res.end(), identity{});
+    }
+
```

Comment:
Thinking about this a bit more I would love to have an overload that separately checks the zero rank case 
```suggestion
    _CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI static constexpr bool
    __check_index(_Extents const& exts) const {
      static_assert(_Extents::rank() == 0, "Rank must be zero for access without indices");
      return true;
    }

    template <class... _SizeTypes>
    _CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI static constexpr bool
    __check_index(_Extents const& exts, _SizeTypes... __indices)
    {
      // ternary avoids "pointless comparison of unsigned integer with zero" warning
      bool __res[ sizeof...(_SizeTypes)]{
        ((is_unsigned_v<index_type> ? true : static_cast<index_type>(__indices) >= 0)
         && static_cast<index_type>(__indices) < exts.extent(_Idxs))...};
      return _CUDA_VSTD::all_of(__res.begin(), __res.end(), identity{});
    }

```

Reply:
We may be able to just fold in a few weeks of time:

```suggestion

    template <class... _SizeTypes>
    _CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI static constexpr bool
    __check_index(_Extents const& exts, _SizeTypes... __indices)
    {
      return
        (((is_unsigned_v<index_type> ? true : static_cast<index_type>(__indices) >= 0)
         && static_cast<index_type>(__indices) < exts.extent(_Idxs))) && ...);
    }

```
which is my preferred solution.

Reply:
Honestly, I would prefer the later one. It is much less code, more readable and easy to understand.